sensor.py

```diff
@@ -106,3 +123,95 @@ def icon(self):
     @property
     def device_class(self):
         return DEVICE_CLASS_BATTERY
+
+
+class HSTemperature(HSSensor):
+    """Representation of a HomeSeer temperature sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                return "°F"
+            elif str(self._units) == 'metric':
+                return "°C"
+            else:
+                return "?"
+        else:
+            return '?'
+
+    @property
+    def icon(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                if self.state < 50:
+                    return 'mdi:thermomter-low'
+                elif self.state > 90:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            elif str(self._units) == 'metric':
+                if self.state < 10:
+                    return 'mdi:thermomter-low'
+                elif self.state > 33:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            else:
+                return 'mdi:thermometer'
+        else:
+            return 'mdi: thermometer'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_TEMPERATURE
+
+
+class HSHumidity(HSSensor):
+    """Representation of a HomeSeer humidity sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        return '%'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_HUMIDITY
+
+
+class HSLuminance(HSSensor):
+    """Representation of a HomeSeer light level sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        return '%'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_ILLUMINANCE
+
+
+class HSFanState(HSSensor):
+    """Representation of a HomeSeer HVAC fan state sensor device."""
+
+    @property
+    def icon(self):
+        if self.state == 0:
+            return 'mdi:fan-off'
+        else:
+            return 'mdi:fan'
+
+
+class HSOperatingState(HSSensor):
+    """Representation of a HomeSeer HVAC operating state sensor device."""
+
+    @property
+    def icon(self):
+        if self.state == 0:         # Idle
+            return 'mdi:fan-off'
+        elif self.state == 1:       # Heating
+            return 'mdi:flame'
+        elif self.state == 2:       # Cooling
+            return 'mdi:snowflake'
+        else:                       # Fan Only
+            return 'mdi:fan'
```

Comment:
We should override the state property of the base class to translate the numeric values to strings like 'idle', 'heat', 'cool', and 'fan' for a better representation in the frontend.

Reply:
I will add that.

Reply:
I think this should be done for 'fan state' as well even though the icon is only off or on. 

thoughts?

Reply:
Yes what are the values of fan state? 0 means off, 8 means high, other values are in between? As long as the representation in HS is consistent, then we should use the strings rather than numeric values.